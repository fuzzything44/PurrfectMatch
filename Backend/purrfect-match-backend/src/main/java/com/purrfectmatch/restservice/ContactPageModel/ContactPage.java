package com.purrfectmatch.restservice.ContactPageModel;

import lombok.Data;

import org.springframework.stereotype.Component;

@Data
@Component
public class ContactPage {
    private String firstName;
    private String lastName;
    private String subject;
    private String email;

    private ShelterEmployee shelterEmployee;
    private String message;

    public enum ShelterEmployee {
        YES, NO
    }

    ContactPage() {
    }

    public ContactPage(String firstName, String lastName, String subject, String email, ShelterEmployee shelterEmployee,
            String message) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.subject = subject;
        this.email = email;
        this.shelterEmployee = shelterEmployee;
        this.message = message;
    }

    @Override
    public String toString() {

        return String.format("First Name:%s\nLastName:%s \nEmail:%s\nShelterEmployee:%s\nMessage:%s", firstName,
                lastName, email, shelterEmployee, message);

    }
}
