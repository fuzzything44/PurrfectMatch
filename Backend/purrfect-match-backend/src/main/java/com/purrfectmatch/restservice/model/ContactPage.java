package com.purrfectmatch.restservice.model;
// package com.purrfectmatch.restservice;

import org.springframework.stereotype.Component;

@Component
public class ContactPage {
    private String firstName;
    private String lastName;
    private String subject;
    private String email;
    private boolean shelterEmployee;
    private String message;

    ContactPage() {}
    
    public ContactPage(String firstName, String lastName, String subject, 
        String email, boolean shelterEmployee, String message) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.subject = subject;
        this.email = email;
        this.shelterEmployee = shelterEmployee;
        this.message = message;
    }

	public String getName() {
		return this.firstName + " " + this.lastName;
	}

	public void setName(String name) {
		String[] parts = name.split(" ");
		this.firstName = parts[0];
		this.lastName = parts[1];
    }
    
    public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String name) {
		String[] parts = name.split(" ");
		this.firstName = parts[0];
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String name) {
        String[] parts = name.split(" ");
		this.lastName = parts[1];
		
	}
    
    public String getSubject() {
		return subject;
	}

	public void setSubject(String subject) {
		this.subject = subject;
    }
    
    public String getEmailAddress() {
        return email;
    }

    public void setEmailAddress(String email) {
        this.email = email;
    }

    public boolean getShelterEmployee() {
        return shelterEmployee;
    }

    public void setShelterEmployee(boolean shelterEmployee) {
        this.shelterEmployee = shelterEmployee;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @Override
    public String toString() {
        return "ContactInfo {" +
            ", firstName='" + firstName  + '\'' +
            ", lastName='" + '\'' +
            ", subject='" + subject + '\'' +
            ", email='" + email + '\'' +
            ", shelterEmployee='" + shelterEmployee + '\'' +
            ", message='" + message + '\'' +
            '}';
    }
}
