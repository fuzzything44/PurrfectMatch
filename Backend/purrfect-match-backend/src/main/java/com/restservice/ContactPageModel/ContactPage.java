package com.restservice.ContactPageModel;

import org.springframework.stereotype.Component;


@Component
public class ContactPage {
    private String firstName;
    private String lastName;
    private String subject;
    private String email;
    private String shelterEmployee;
    private String message;

    ContactPage() {
    }

    public ContactPage(String firstName, String lastName, String subject, String email, String shelterEmployee,
            String message) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.subject = subject;
        this.email = email;
        this.shelterEmployee = shelterEmployee;
        this.message = message;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String name) {
        ;
        this.firstName = name;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String name) {
        this.lastName = name;

    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getShelterEmployee() {
        return shelterEmployee;
    }

    public void setShelterEmployee(String shelterEmployee) {
        this.shelterEmployee = shelterEmployee;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @Override
    public String toString() {

        return String.format("First Name:%s\nLastName:%s \nEmail:%s\nShelterEmployee:%s\nMessage:%s",firstName,lastName,email,shelterEmployee,message);
       
        
    }
}
