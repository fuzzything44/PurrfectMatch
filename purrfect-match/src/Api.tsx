import { SuccessStoryInfo, SUCCESS_STORY_PAGE_SIZE } from "./Definitions";

/*
    This class is used for whenever we need to get data from the server. 
*/
export class Api {
    public static getSuccessStories = (breakpoint?: number): Promise<SuccessStoryInfo[]> => {
        // The call to the server itself will be paginated, so which page of stories we want.
        // No value = first page. 
        if (breakpoint === undefined) {
            breakpoint = 0;
        }

        // This is pretty much all temporary code that will be replaced when we actually communicate with the server. 
        const ALL_SUCCESS_STORIES: SuccessStoryInfo[] = [
            {
                image: "cat.jpg",
                name: "Mittens",
                age: 4,
                storyText: "Some story about how this cat was saved from horrible living conditions and is doing much better now."
            },
            {
                image: "Cat_1.png",
                name: "Fluffy",
                age: 36,
                storyText: "Another success story here"
            },
            {
                image: "Dog_1.png",
                name: "Mr. Woof",
                age: 60,
                storyText: "Okay, success stories really should be longer but eh. "

            }
        ];
        let fakeApiResult = ALL_SUCCESS_STORIES.slice(SUCCESS_STORY_PAGE_SIZE * breakpoint, SUCCESS_STORY_PAGE_SIZE * breakpoint + SUCCESS_STORY_PAGE_SIZE);
        return Promise.resolve(fakeApiResult);
    }
}