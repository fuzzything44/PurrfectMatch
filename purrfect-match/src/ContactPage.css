.textBox {
    background-color: #D1F9FE;
    border-color: black;
    background: solid;
    border: solid;
    border-radius: 5px;
    margin: 5px;
}
#send{
    background-color: lightgrey;
    border-color: black;
    background: solid;
    text-align: center;
    display: inline-block;
    border: none;
    width: 3.3em;
    height: 1.4em;
    border-radius: 3px;
}

#send:hover {background-color: grey}

.textBox.small {
    position: relative;
    width: 10em;
    height: 1.3em;
    border-radius: 5px;
}

.textBox.message{
    position: relative;
    width: 30em;
    height: 15em;
    border-radius: 10px;
    margin: 10px;
}

.row{
    margin-left: 1 em;
}

.float{
    float: left;
}