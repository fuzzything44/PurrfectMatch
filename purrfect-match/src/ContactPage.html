<!doctype html>
<html lang="us">
<link href="Contact Us Page.css" rel="stylesheet"></link>
<div class="row">
    First Name 
    <div>
        <input type="text" name="fname" id="textBox" class="small"/>
    </div>
</div>
<div class="row">
    Last Name
    <div>
        <input type="text" name="lname" id="textBox" class="small"/>
    </div>
</div>
<div class="row">
    Subject
    <div>
        <input type="text" name="subject" id="textBox" class="small"/>
    </div>
<brk>
</brk>
<div class="row">
    Email
    <div>
        <input type="text" name="email" id="textBox" class="small"/>
    </div>
</div>
<div class="row">
    Are you a shelter employee?
    <div>
        <input type="text" name="ifEmployee" id="textBox" class="small"/>
    </div>
</div>
<div class="row">
    Message:
    <div>
        <textarea name="message" rows="20" cols="40" id="textBox" class="message">Type your message here...
        </textarea>
    </div>
</div>
<button id="send">Send</button>