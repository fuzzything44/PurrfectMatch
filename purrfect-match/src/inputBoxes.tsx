import React from 'react';
import "./ContactPage.css"

interface inputProps {
    prompt: string;
}

export class Input extends React.Component<inputProps, {}> {
    
    public render() {
        return (
        <div className="row">
            {this.props.prompt} 
            <div>
                <input type="text" name="fname" className="textBox"/>
            </div>
        </div>
        );
    }
}

export class Message extends React.Component<inputProps, {}> {
    
    public render() {
        return (
        <div className="row">
            {this.props.prompt} 
            <div>
                <textarea name="message" rows={20} cols={40} className="textBox message" placeholder = "Type your message here...">
                </textarea>
            </div>
        </div>
        );
    }
}

export class Send extends React.Component<{},{}> {

    public render() {
        return (
            <button id="send"> Send</button>
        );
    }
}