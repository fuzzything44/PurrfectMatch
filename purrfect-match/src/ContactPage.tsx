import React from 'react';
import './ContactPage.css';
import { Input, Select, Message, Send } from './inputBoxes';
import { Api } from "Api";
import Footer from 'Footer/Footer';
interface ContactPageProps {
};

interface ContactPageState {
    firstName: string | undefined;
    lastName: string | undefined;
    subject: string | undefined;
    email: string | undefined;
    shelterEmployee: "Yes" | "No" | "" | undefined;
    message: string | undefined;
};

const OPTION_ARRAY = ["Yes", "No"];

export class ContactPage extends React.Component<ContactPageProps, ContactPageState> {

    constructor(props: any) {
        super(props);

        this.state = {
            firstName: undefined,
            lastName: undefined,
            subject: undefined,
            email: undefined,
            shelterEmployee: undefined,
            message: undefined
        };
    }

    fillAndValidateField = (field: keyof ContactPageState): boolean => {
        if (this.state[field] === undefined) {
            this.setState({
                [field]: ""
            } as any);
            return false;
        } else {
            return this.state[field] != "";
        }
    }

    handleSubmit = () => {
        let formValid = true;
        // Validate all fields filled
        for (let elem of ["firstName", "lastName", "subject", "email", "shelterEmployee", "message"]) {
            if (!this.fillAndValidateField(elem as keyof ContactPageState)) {
                formValid = false;
            }
        }
        if (formValid) {
            // @ts-ignore: Strict null checks, but we know they're non-null from the validation we just did
            Api.submitContactInfo(this.state.firstName, this.state.lastName, this.state.subject, this.state.email, this.state.shelterEmployee, this.state.message);
        }
    }

    checkField = (fieldVal: string | undefined, error: string): string | undefined => {
        if (fieldVal == "") {
            return error;
        }
        return undefined;
    }

    public render() {
        return (
            <div className="contactPage">
                <h2 className="header">Contact Us!</h2>
                <div className="contactBoxes">
                    <Input
                        prompt="First Name:"
                        error={this.checkField(this.state.firstName, "Provide your first name.")}
                        onchange={(newValue: string) => this.setState({ firstName: newValue })}
                    />
                    <Input
                        prompt="Last Name:"
                        error={this.checkField(this.state.lastName, "Provide your last name.")}
                        onchange={(newValue: string) => this.setState({ lastName: newValue })}
                    />
                    <Input
                        prompt="Subject:"
                        error={this.checkField(this.state.subject, "Provide a subject.")}
                        onchange={(newValue: string) => this.setState({ subject: newValue })}
                    />
                    <Input
                        prompt="Email Address:"
                        error={this.checkField(this.state.email, "Provide your email.")}
                        onchange={(newValue: string) => this.setState({ email: newValue })}
                    />
                </div>
                <div className="contactBoxes">
                    <div className="contactText">
                        <p>
                            Are you a shelter and need help with your account?<br />
                            Are you a potential pet owner with comments or<br />
                            concerns? Send us a note here! <br />
                            Or you can reach us at purrfectmatch@gmail.com
                        </p>
                    </div>
                    <Select
                        prompt="Are you a shelter employee?"
                        options={OPTION_ARRAY}
                        error={this.checkField(this.state.shelterEmployee, "Choose an option.")}
                        onchange={(newValue: any) => this.setState({ shelterEmployee: newValue })}
                    />
                </div>
                <div className="contactBoxes">
                    <Message
                        prompt="Message:"
                        error={this.checkField(this.state.message, "Please write a message.")}
                        onchange={(newValue: string) => this.setState({ message: newValue })}
                    />
                </div>
                <Send onclick={this.handleSubmit} />
            </div>
        );
    }
}
