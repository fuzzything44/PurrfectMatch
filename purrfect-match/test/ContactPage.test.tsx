import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ContactPage } from 'ContactPage';
import { Input, Select, Message, Send } from 'inputBoxes';
import { Api } from 'Api';

configure({ adapter: new Adapter() });

jest.mock("Api");
jest.mock("inputBoxes");

let wrapper: ShallowWrapper<any, any, any>;
beforeEach(() => {
    wrapper = shallow(<ContactPage />);
});

test("Renders correctly", () => {
    expect(wrapper).toMatchSnapshot();

    // Setup second snapshot (with errors shown)
    wrapper.setState({
        firstName: "",
        lastName: "",
        subject: "",
        email: "",
        shelterEmployee: "",
        message: ""
    });
    expect(wrapper).toMatchSnapshot();
});

test("Submits correct information", () => {
    // Add some mock information. 
    wrapper.setState({
        firstName: "mockName",
        lastName: "mockLast",
        subject: "mockSubj",
        email: "mockEmail@website.com",
        shelterEmployee: "No",
        message: "mockMessage"
    });
    // Setup response from Api call
    (Api.submitContactInfo as jest.MockedFunction<() => Promise<void>>).mockReturnValueOnce(Promise.resolve());

    // Click submit button
    expect(wrapper.find(Send).props().onclick).not.toBeUndefined();
    // @ts-ignore: We know onclick() is defined.
    wrapper.find(Send).props().onclick();

    // At this point, Api.submitContactInfo() should have been called with our provided data.
    expect(Api.submitContactInfo).toHaveBeenCalledWith("mockName", "mockLast", "mockSubj", "mockEmail@website.com", "No", "mockMessage");
});

test("Information %s propogates to page correctly", () => {
    // Given an element and the specific field, this checks if changing that element updates the field
    function checkPropagation(onchange: ((newVal: string) => void) | undefined, field: keyof ContactPage["state"]) {
        // Make sure not undefined
        expect(onchange).not.toBeUndefined();
        // Since not undefined, we're safe to cast that away and call it.
        (onchange as (newVal: string) => void)("updatedValue-" + field);
        // Verify that the main component updated
        expect(wrapper.state(field)).toBe("updatedValue-" + field);
    }

    // Input box checks
    checkPropagation(wrapper.find(Input).at(0).props().onchange, "firstName");
    checkPropagation(wrapper.find(Input).at(1).props().onchange, "lastName");
    checkPropagation(wrapper.find(Input).at(2).props().onchange, "subject");
    checkPropagation(wrapper.find(Input).at(3).props().onchange, "email");

    // Dropdown check
    checkPropagation(wrapper.find(Select).at(0).props().onchange, "shelterEmployee");

    // Text area check
    checkPropagation(wrapper.find(Message).at(0).props().onchange, "message");
});

test("fillAndValidateField fills undefined field with \"\" and returns false", () => {
    expect(wrapper.instance().fillAndValidateField("firstName")).toBeFalsy();
    expect(wrapper.state("firstName")).toBe("");
});

test("fillAndValidateField returns false on empty string fields", () => {
    wrapper.setState({ firstName: "" });
    expect(wrapper.instance().fillAndValidateField("firstName")).toBeFalsy();
    expect(wrapper.state("firstName")).toBe("");
});

test("fillAndValidateField does not change filled fields and returns true", () => {
    wrapper.setState({ firstName: "mockFirst" });
    expect(wrapper.instance().fillAndValidateField("firstName")).toBeTruthy();
    expect(wrapper.state("firstName")).toBe("mockFirst");
});

test("handleSubmit does not make API call if validation fails", () => {
    wrapper.instance().fillAndValidateField = jest.fn()
        .mockReturnValueOnce(true)  // firstName
        .mockReturnValueOnce(false) // lastName
        .mockReturnValueOnce(true)  // subject
        .mockReturnValueOnce(true)  // email
        .mockReturnValueOnce(true)  // shelterEmploye
        .mockReturnValueOnce(true)  // message
    wrapper.instance().handleSubmit();

    expect(wrapper.instance().fillAndValidateField).toHaveBeenCalledTimes(6);
    expect(Api.submitContactInfo).not.toHaveBeenCalled();
});

test("checkField returns correct error message or nothing", () => {
    expect(wrapper.instance().checkField(undefined, "mockErr")).toBeUndefined();
    expect(wrapper.instance().checkField("mockData", "mockErr")).toBeUndefined();
    expect(wrapper.instance().checkField("", "mockErr")).toBe("mockErr");
});
