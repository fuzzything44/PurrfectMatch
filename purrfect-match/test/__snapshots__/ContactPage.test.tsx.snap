// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="contactPage"
>
  <h2
    className="header"
  >
    Contact Us!
  </h2>
  <div
    className="contactBoxes"
  >
    <Input
      onchange={[Function]}
      prompt="First Name:"
    />
    <Input
      onchange={[Function]}
      prompt="Last Name:"
    />
    <Select
      onchange={[Function]}
      options={
        Array [
          "Yes",
          "No",
        ]
      }
      prompt="Are you a shelter employee?"
    />
    <Input
      onchange={[Function]}
      prompt="Email Address:"
    />
    <Input
      onchange={[Function]}
      prompt="Subject:"
    />
  </div>
  <div
    className="contactBoxes"
  >
    <div
      className="contactText"
    >
      <p>
        Are you a shelter and need help with your account?
        <br />
        Are you a potential pet owner with comments or
        <br />
        concerns? Send us a note here! 
        <br />
        Or you can reach us at purrfectmatch@gmail.com
      </p>
    </div>
    <div
      className="contactBoxes"
    >
      <Message
        onchange={[Function]}
        prompt="Message:"
      />
    </div>
  </div>
  <Send
    className="contactPageButton"
    onclick={[Function]}
  />
  <Modal
    display={false}
    onClose={[Function]}
  >
    <div
      className="modalSpacer"
    />
  </Modal>
</div>
`;

exports[`Renders correctly 2`] = `
<div
  className="contactPage"
>
  <h2
    className="header"
  >
    Contact Us!
  </h2>
  <div
    className="contactBoxes"
  >
    <Input
      error="Provide your first name."
      onchange={[Function]}
      prompt="First Name:"
    />
    <Input
      error="Provide your last name."
      onchange={[Function]}
      prompt="Last Name:"
    />
    <Select
      error="Choose an option."
      onchange={[Function]}
      options={
        Array [
          "Yes",
          "No",
        ]
      }
      prompt="Are you a shelter employee?"
    />
    <Input
      error="Provide a valid email."
      onchange={[Function]}
      prompt="Email Address:"
    />
    <Input
      error="Provide a subject."
      onchange={[Function]}
      prompt="Subject:"
    />
  </div>
  <div
    className="contactBoxes"
  >
    <div
      className="contactText"
    >
      <p>
        Are you a shelter and need help with your account?
        <br />
        Are you a potential pet owner with comments or
        <br />
        concerns? Send us a note here! 
        <br />
        Or you can reach us at purrfectmatch@gmail.com
      </p>
    </div>
    <div
      className="contactBoxes"
    >
      <Message
        error="Please write a message."
        onchange={[Function]}
        prompt="Message:"
      />
    </div>
  </div>
  <Send
    className="contactPageButton"
    onclick={[Function]}
  />
  <Modal
    display={false}
    onClose={[Function]}
  >
    <div
      className="modalSpacer"
    />
  </Modal>
</div>
`;
